src/python/WMCore/WMBS/MySQL/Locations/GetSiteInfo.py

```diff
@@ -29,4 +29,30 @@ def execute(self, siteName=None, conn=None, transaction=False):
             sql = self.sql + " WHERE wl.site_name = :site"
             results = self.dbi.processData(sql, {'site': siteName},
                                        conn=conn, transaction=transaction)
-        return self.formatDict(results)
+        return self.format(results)
+
+    def format(self, result):
+        """
+        Format the DB results in a plain list of dictionaries, with one
+        dictionary for each site name, thus with a list of PNNs.
+        :param result: DBResult object
+        :return: a list of dictionaries
+        """
+        # first create a dictionary to make key look-up easier
+        resp = {}
+        for thisItem in DBFormatter.format(self, result):
```

Comment:
since below you takes elements of `thisItem` it is better to check that its length has at least 7 elements (the largest index I see in code below). If not it is better to provide appropriate logging message and skip the rest of the  code in a loop.

Reply:
It could be done, but this is a SQL statement where we request 7 fields, hence it must have 7 elements in the data objects. I don't think we gain anything from adding this check here.

Reply:
Alan, my point is that it was unclear reading the PR code why we need 7 elements until I opened full file and saw the SQL statement which you mentioned. If you do not want to perform the check which will not hurt the codebase, then at least put appropriate comment either in docstring or better directly in for loop codebase that severn elements are part of SQL statement.